nifi-nar-bundles/nifi-asn1-bundle/nifi-asn1-services/src/main/java/org/apache/nifi/jasn1/JASN1Reader.java

```diff
@@ -134,10 +135,35 @@ public class JASN1Reader extends AbstractConfigurableComponent implements Record
         .required(false)
         .build();
 
+    private static final PropertyDescriptor DO_ADDITIONAL_PREPROCESSING = new PropertyDescriptor.Builder()
+        .name("do-additional-preprocessing")
+        .displayName("Do Additional Preprocessing")
```

Comment:
This property name should be changed to something more descriptive. Starting with `Do` seems a bit too casual for a property name.

Instead of having this as a Boolean property, what do you think about naming it `Schema Preparation Strategy`? With a default value of `DISABLED` and and allowable value of `PREPROCESSED`? This would help communicate what the property does and leave the option open for alternative strategies.

Reply:
We can go this route. However I think we should remain consistent with our communication in that the documentation already discusses pre-processing, namely parsing and compiling the schema.
That's why I went with the "additional" prefix.
I'll add something that incorporates both ideas.